### PR TITLE
Restructure SirCAL to use API data for lag calculation

### LINKED ISSUE #862: Switch SirCAL from metadata to API data for lag
Dashboard viz use the raw API data which means alerts and dashboards can disagree if metadata is out of sync with API data. Switching to use API data should remove this problem.

### PR DESCRIPTION
**Pull request by benjaminysmith**
### Description
Restructure SirCAL to use API data for lag calculation

### Changelog
- Move API retrieval up so that it runs for all signals
- Use the API lag data to generate alert (instead of the metadata lag)

### Fixes 
- Fixes #862

### CODE REVIEW

File: sir_complainsalot/delphi_sir_complainsalot/check_source.py [hunk 1]
```diff
@@ -103,20 +83,37 @@ def check_source(data_source, meta, params, grace, logger):
...
             continue
 
-        # convert numpy datetime values to pandas datetimes and then to
-        # datetime.date, so we can work with timedeltas after
         unique_dates = [pd.to_datetime(val).date()
                         for val in latest_data["time_value"].unique()]
-        unique_issues = [pd.to_datetime(val).date()
-                        for val in latest_data["issue"].unique()]
-
         current_lag_in_days = (datetime.now().date() - max(unique_dates)).days
```
Comment (krivard) [inline sir_complainsalot/delphi_sir_complainsalot/check_source.py:88]: what happens if unique_dates is empty? `google_symptoms` has no data from the last 14 days.

Comment (benjaminysmith) [reply sir_complainsalot/delphi_sir_complainsalot/check_source.py]: Hmmm good point. Ok, wdyt about this version where we fallback to using metadata (and flag this in the log) if we get no data from the API?

Output using the current prod config is shown below:

{"message": "is 77 days old - (last update: 2020-12-08)", "data_source": "google-symptoms", "signal_and_geo_types": "`ageusia_raw_search: [county, hhs, hrr, msa, nation, state]`\n`ageusia_smoothed_search: [county, hhs, hrr, msa, nation, state]`\n`anosmia_raw_search: [county, hhs, hrr, msa, nation, state]`\n`anosmia_smoothed_search: [county, hhs, hrr, msa, nation, state]`\n`sum_anosmia_ageusia_raw_search: [county, hhs, hrr, msa, nation, state]`\n`sum_anosmia_ageusia_smoothed_search: [county, hhs, hrr, msa, nation, state]`\n", "event": "Signal out of SLA", "logger": "delphi_sir_complainsalot.run", "level": "critical", "timestamp": "2021-02-23T17:59:35.978353Z"}

Comment (krivard) [reply sir_complainsalot/delphi_sir_complainsalot/check_source.py]: perfect! for signals that haven't updated in two weeks, it won't matter if meta is a day behind

**Review by krivard — APPROVED** [review]
xlnt

**Pull request merged**